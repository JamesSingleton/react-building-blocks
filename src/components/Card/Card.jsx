import React from 'react';
import styled, { css, keyframes } from 'styled-components';

const animatedCss = css`
  opacity: 1;
  transform: translateY(0);
`;

const primaryCss = css`
  background-color: #008bf8;
  color: #fff;
`;

const StyledCard = styled.div`
  width: ${props => (props.big ? '450px' : '300px')};
  padding: 15px;
  opacity: 0;
  transform: translateY(50px);
  transition: 500ms all ease-in-out;
  margin: ${props => (props.noMargin ? 0 : '15px')};
  box-shadow: 0 5px 15px -5px rgba(0, 0, 0, 1);
  border-radius: 5px;
  ${props => props.animated && animatedCss}
  ${props => props.primary && primaryCss}
`;

class Card extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      animated: false
    };
  }

  componentDidMount() {
    const { delay } = this.props;
    setTimeout(() => {
      this.setState(() => ({ animated: true }));
    }, delay);
  }

  render() {
    const {
      delay = 0,
      noAnimation,
      primary,
      noMargin,
      big,
      ...props
    } = this.props;
    const { animated } = this.state;
    return (
      <StyledCard
        animated={animated}
        delay={delay}
        primary={primary}
        noAnimation={noAnimation}
        big={big}
        noMargin={noMargin}
        {...props}
      />
    );
  }
}

export default Card;
