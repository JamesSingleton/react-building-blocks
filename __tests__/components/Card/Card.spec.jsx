import React from 'react';
import { mount } from 'enzyme';
import Card from '../../../src/components/Card';

describe('Card', () => {
  it('should renders the Card', () => {
    const renderedCard = mount(<Card><div>Hello Span</div></Card>);
    expect(renderedCard).toMatchSnapshot();
  });
  it('Stretches Vertically', () => {
    const props = {
      stretchVertically: true,
    };
    const renderedCard = mount(<Card {...props}><div>Hello Span</div></Card>);
    expect(renderedCard).toMatchSnapshot();
  });
});
