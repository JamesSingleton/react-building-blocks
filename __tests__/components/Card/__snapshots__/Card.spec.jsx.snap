// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Stretches Vertically 1`] = `
<Card
  className=""
  hideOverflow={true}
  stretchVertically={true}
>
  <styled.div
    className=""
    hideOverflow={true}
    stretchVertically={true}
  >
    <div
      className="sc-AykKC jxtHeF"
    >
      <div>
        Hello Span
      </div>
    </div>
  </styled.div>
</Card>
`;

exports[`Card should renders the Card 1`] = `
<Card
  className=""
  hideOverflow={true}
  stretchVertically={false}
>
  <styled.div
    className=""
    hideOverflow={true}
    stretchVertically={false}
  >
    <div
      className="sc-AykKC iIMINq"
    >
      <div>
        Hello Span
      </div>
    </div>
  </styled.div>
</Card>
`;
